### PR TITLE
[OYPD-408] Anonymous should not see Session, Alert, Activity, or Memb…

### PR DESCRIPTION
**Pull request by Kerby70**
…ership pages; and should only see Branch pages that are not coming soon.

Jira https://propeople-us.atlassian.net/browse/OYPD-408
Github task https://github.com/ymcatwincities/openy/issues/199
D.o https://www.drupal.org/node/2877684

Steps to review
- [x] Get urls for Session, Alert, Activity, Membership pages, and Branch pages, include 1 branch with coming soon & one without.
- [x] As anonymous
- [x] Should not be able to see pages for Session, Alert, Activity, or Membership pages
- [x] Should only see Branch pages that are **not** coming soon

Make sure these boxes are checked before asking for review of your pull request - thank you!

- [x] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI.
- [x] All tests are running and there are no failed tests reported by CI.
- [x] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [x] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes.
- [x] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [x] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there.
- [x] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1155 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1155 |
| Installation process | http://install.openy-dev.ffwua.com/build1155/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1155/

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1155/profiles/contrib/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1156 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1156 |
| Installation process | http://install.openy-dev.ffwua.com/build1156/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1156/

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1157 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1157 |
| Installation process | http://install.openy-dev.ffwua.com/build1157/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1157/

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1156/profiles/contrib/openy/build/reports/behat

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1157/profiles/contrib/openy/build/reports/behat

**Comment by ivan-berezhnov** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1163 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1163 |
| Installation process | http://install.openy-dev.ffwua.com/build1163/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1163/

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1163/profiles/contrib/openy/build/reports/behat

File: composer.json [hunk 1]
```diff
@@ -171,6 +171,7 @@
         "drupal/simple_sitemap": "^2.8",
         "drupal/redirect": "^1.0@alpha",
         "drupal/metatag": "^1.0",
-        "drupal/google_tag": "1.x-dev"
+        "drupal/google_tag": "1.x-dev",
+        "drupal/rabbit_hole": "1.x-dev"
```
Comment (alexschedrov) [inline composer.json]: In drush-make file we have `1.0-beta3`, but here `1.x-dev`.

File: openy.info.yml [hunk 2]
```diff
@@ -70,6 +70,8 @@ dependencies:
   - blazy
   - blazy_ui
   - geolocation
+  - rabbit_hole
+  - rh_node
```
Comment (alexschedrov) [inline openy.info.yml:74]: We should explicitly enable those modules in hook_update in `openy.install`.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by alexschedrov** [comment]
1. This functionality doesn't work on upgrade path environment.
Prlbably because rh_node module is not enabled.
image

screen shot 2017-05-12 at 11 54 31 am

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1168 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1168 |
| Installation process | http://install.openy-dev.ffwua.com/build1168/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1168/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://ci.openymca.org/build1168/profiles/contrib/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]
Approved.

**Pull request merged**